daemon/container_operations_unix.go

```diff
@@ -539,24 +562,10 @@ func (daemon *Daemon) allocateNetwork(container *container.Container) error {
 
 	updateSettings := false
 	if len(container.NetworkSettings.Networks) == 0 {
-		mode := container.HostConfig.NetworkMode
-		if container.Config.NetworkDisabled || mode.IsContainer() {
```

Comment:
By moving the functionality into `UpdateNetworkSettings`, this is a case where we must handle properly.
Else, we will end up processing the allocateNetwork for `NetworkDisabled || IsContainer()` case, which is incorrect.